conans/test/generators/pkg_config_test.py

```diff
@@ -55,3 +58,94 @@ def variables_setup_test(self):
 Libs: -L${libdir}
 Cflags: -I${includedir} -Flag1=23 -DMYDEFINE1
 """)
+
+    def pkg_config_dirs_test(self):
+        # https://github.com/conan-io/conan/issues/2756
+        conanfile = """
+import os
+from conans import ConanFile
+
+class PkgConfigConan(ConanFile):
+    name = "MyLib"
+    version = "0.1"
+
+    def package_info(self):
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.bindirs = []
+        self.cpp_info.libs = []
+        libname = "mylib"
+        fake_dir = os.path.abspath(os.path.join("my_absoulte_path", "fake"))
```

Comment:
The test is ok, but this package will be un-reusable. 
The thing you have to test is: If you get an absolute path but built from the `self.package_folder` is prefixed ok in the pkg file, never containing an abs path there.